dev_scripts_helpers/thin_client/test/test_setenv.py

```diff
@@ -0,0 +1,231 @@
+# _SetenvTestCase
+# #############################################################################
+
+
+class _SetenvTestCase(hunitest.TestCase):
+    """
+    Helper test class to perform common setup and provide utility methods for
+    testing setenv.sh script.
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup_teardown_test(self) -> None:
+        self.set_up_test()
+
+    def set_up_test(self) -> None:
+        """
+        Set up test environment.
+        """
+        # Get the path to the `setenv.sh` script.
+        git_root = hgit.find_git_root()
+        setenv_files = hsystem.find_file_with_dir(
+            "thin_client/setenv.sh",
+            root_dir=git_root,
+            dir_depth=1,
+            mode="assert_unless_one_result",
+        )
+        self.setenv_path = pathlib.Path(setenv_files[0])
+        _LOG.debug("setenv_path=%s", self.setenv_path)
+        # Get the path to the `thin_client_utils.sh` and `repo_config.yaml`
+        # files that are used by `setenv.sh`.
+        utils_files = hsystem.find_file_with_dir(
+            "thin_client/thin_client_utils.sh",
+            root_dir=git_root,
+            dir_depth=1,
+            mode="assert_unless_one_result",
+        )
+        self.thin_client_utils_path = pathlib.Path(utils_files[0])
+        _LOG.debug("thin_client_utils_path=%s", self.thin_client_utils_path)
+        self.repo_config_path = pathlib.Path(
+            hsystem.find_file_in_repo("repo_config.yaml", root_dir=git_root)
+        )
+        _LOG.debug("repo_config_path=%s", self.repo_config_path)
+        # Exit if files do not exist.
+        hdbg.dassert_file_exists(str(self.setenv_path))
+        hdbg.dassert_file_exists(str(self.thin_client_utils_path))
+        hdbg.dassert_file_exists(str(self.repo_config_path))
+        _LOG.debug(
+            "Initialized _SetenvTestCase with setenv.sh, thin_client_utils.sh, and repo_config.yaml"
+        )
+
+    def run_setenv_in_clean_bash(self) -> Tuple[int, str]:
+        """
+        Run setenv.sh in a clean isolated bash environment, capturing both
+        output and environment variables.
+
+        :return: tuple of (return_code, output)
+        """
+        wrapper_script = self._create_wrapper_script()
+        # Create a temporary executable file.
+        tmp_dir = self.get_scratch_space()
+        wrapper_path = os.path.join(tmp_dir, "tmp_wrapper.sh")
+        hio.create_executable_script(wrapper_path, wrapper_script)
+        _LOG.debug("Running setenv.sh in clean bash environment")
+        result = hsystem.system_to_string(
+            f"env -i bash {wrapper_path}",
+            abort_on_error=False,
+            log_level=logging.DEBUG,
+        )
+        return result
+
+    def _create_wrapper_script(self) -> str:
+        """
+        Create a wrapper script to run setenv.sh in a clean isolated bash
+        environment.
+
+        :return: wrapper script content
+        """
+        wrapper_script = f"""#!/bin/bash
```

Comment:
You can indent and then use hprint.dedent to realigned.

It's difficult to read when things are aligned according to the "script" and not to the "code"